### PR TITLE
Multimedia model 

### PR DESCRIPTION
**Pull request by alexlib**
Multimedia model includes the following functions 
init_mmLUT
get_mmf_from_mmLUT
trans_Cam_Point
back_trans_Point
volumedimension

### CODE REVIEW

**Comment by yosefm** [comment]
First thing I see: I think the treatment of mmLUT is very problematic. It would make much more sense to extend the Calibration object to contain this camera-specific information. Then we won't have to pass around a camera number, and the functions will be able to directly access the relevant info from the Calibration they get.

File: liboptv/include/multimed.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+} mmlut;
+
+
+
+double get_mmf_from_mmLUT (int i_cam, double X, double Y, double Z, mmlut *mmLUT);
+
+/* Note that multimed_nlay_v2 is renamted to _nlay) */
+void  multimed_nlay (Exterior *ex
+                   , mm_np *mm
+                   , double X
```
Comment (yosefm) [inline liboptv/include/multimed.h]: X,Y,Z should be a vec3d type - everywhere it appears.

File: liboptv/include/multimed.h [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+/* mmLUT structure */
+typedef struct {
+    Origin origin;
+    int    nr, nz, rw;
+    double *data; 
+} mmlut;
+
+
+
+double get_mmf_from_mmLUT (int i_cam, double X, double Y, double Z, mmlut *mmLUT);
```
Comment (yosefm) [inline liboptv/include/multimed.h]: Here and in other functions, I propose to make mmLUT the first argument, like 'self' in Python.

File: liboptv/src/multimed.c [hunk 3]
```diff
@@ -0,0 +1,658 @@
...
+        
+      /* intersect with image vertices rays */
+      
+      for (i = 0; i < 2; i ++) {
+         for (j = 0; j < 2; j++) {
+          
+          pixel_to_metric (&x, &y, xc[i], yc[j], cpar);
+          
+        
+          /* x = x - I[i_cam].xh;
```
Comment (yosefm) [inline liboptv/src/multimed.c]: Remove

**Comment by alexlib** [comment]
I do not understand what to do. mmLUT is a ad-hoc per-camera developed
look-up table to speed up the calculations of transformations from image to
physical space and back through the glass wall.

What do you decide to add to the Calibration object? something like
cal->mmlut ? it totally makes sense, but it will break a lot of things, no?
Shall we have a separate PR on this change?

**Comment by yosefm** [comment]
It won't break anything in liboptv because we are building it from scratch. In openptv-python, everything we bring in from liboptv requires some interface changes. I think it is worth it, we do it only once, clean, and then the future is paradise.

[PS - We need to start bringing in my changes to pyptv, they're getting piled up but I'm only now dogfooding them]

**Comment by alexlib** [comment]
how do we deal with the fact that the size of mmLUT is not known before
it's initialized and it changes as we do calibration?

**Comment by alexlib** [comment]
I've updated the PR accordingly: mmLUT is not the first argument, X,Y,Z are replaced by vec3d

**Comment by yosefm** [comment]
AFAICT, the only thing that changes size is the allocation of nnLUT.data. To be clear, I propose to keep the structure mmlut as member of Calibration, not to move its members into it (although we could possibly trim it, but that can wait). So we'll have cal.lut.data allocated at init, and freed with free_calibration().

**Comment by alexlib** [comment]
something like this?

/* mmLUT structure */
typedef struct {
    Origin origin;
    int    nr, nz, rw;
    double *data;
} mmlut;

typedef struct {
    Exterior ext_par;
    Interior int_par;
    Glass glass_par;
    ap_52 added_par;
    mmlut mmlut
} Calibration;


Please, see the branch I'm working
on f162112b42283cb35ea24df238cf53fe81d8550e

**Pull request closed without merging**